### DISCUSSION TITLE
Does ECJ support value based classes?

### DISCUSSION THREAD

**Discussion by laeubi** [eclipse-jdt/eclipse.jdt.core#2626] [Q&A]
I just came across [JEP 390](https://openjdk.org/jeps/390) and it says:

> Within the JDK, the `@jdk.internal.ValueBased` annotation is used to signal to **javac** and HotSpot that a class is [value-based](https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/doc-files/ValueBased.html)

So the question would be does **ecj batch compiler** also support `@jdk.internal.ValueBased` annotations? So if I would add that to a custom type will it have any influence?

**Comment by stephan-herrmann** [comment]
Please have a look at https://github.com/eclipse-jdt/eclipse.jdt.core/blob/master/org.eclipse.jdt.core.tests.compiler/src/org/eclipse/jdt/core/tests/compiler/regression/ValueBasedAnnotationTests.java

I can't vouch for the support being complete, so have a look at the tests and/or make your own experiments.
